pkg/cnservice/server.go

```diff
@@ -147,33 +155,56 @@ func (s *service) initEngine(
 	return nil
 }
 
-func (s *service) createMOServer(inputCtx context.Context, pu *config.ParameterUnit) {
+const s3FileServiceName = "S3" // same as dnservice/factory.go s3FileServiceName
+
+func (s *service) createMOServer(inputCtx context.Context, pu *config.ParameterUnit) (context.Context, error) {
 	address := fmt.Sprintf("%s:%d", pu.SV.Host, pu.SV.Port)
 	moServerCtx := context.WithValue(inputCtx, config.ParameterUnitKey, pu)
 	s.mo = frontend.NewMOServer(moServerCtx, address, pu)
-	{
-		// init trace/log/error framework
-		if _, err := trace.Init(moServerCtx,
-			trace.WithMOVersion(pu.SV.MoVersion),
-			trace.WithNode("node_uuid", trace.NodeTypeCN),
-			trace.EnableTracer(!pu.SV.DisableTrace),
-			trace.WithBatchProcessMode(pu.SV.TraceBatchProcessor),
-			trace.DebugMode(pu.SV.EnableTraceDebug),
-			trace.WithSQLExecutor(func() ie.InternalExecutor {
-				return frontend.NewInternalExecutor(pu)
-			}),
-		); err != nil {
-			panic(err)
+
+	// init trace/log/error framework
+	var fs fileservice.FileService
```

Comment:
Please call s.getFileService to get the FileService instance.

Reply:
not found `s.getFileService`  funtion, but i could simply use `s.fileService`